pkg/ibm/ibmz_helpers.go

```diff
@@ -287,3 +290,44 @@ func (iz IBMZDynamicConfig) assignIpToInstance(instance *vpcv1.Instance, vpcServ
 	ip, err = assignNewlyAllocatedIP(instance, vpcService)
 	return ip, err
 }
+
+// findInstancesWithoutTaskRuns iterates over instances retrieved from the iz cloud and returns a list of those that
+// are associated with a non-existing Tekton TaskRun. Each instance's volume should have a tag with the associated
+// TaskRun's namespace and name. This is compared to existingTaskRuns, which is a map of namespaces to a list of
+// TaskRuns in that namespace, to determine if this instance's TaskRun still exists.
+func (iz IBMZDynamicConfig) findInstancesWithoutTaskRuns(log logr.Logger, vpcService *vpcv1.VpcV1, instances []vpcv1.Instance, existingTaskRuns map[string][]string) []string {
+	var instancesWithoutTaskRuns []string
+
+	// Iterate over all VM instances
+	for _, instance := range instances {
+		volumeId := instance.VolumeAttachments[0].ID
+		// Get instance's volume; assumes only one volume per instance
+		volume, _, err := vpcService.GetVolume(&vpcv1.GetVolumeOptions{ID: volumeId})
+		if err != nil {
+			msg := fmt.Sprintf("failed to get instance %s's volume; skipping this instance...", *instance.ID)
+			log.Error(err, msg)
+			continue
+		}
+
+		// Try to find the volume's TaskRun ID tag
+		volumeTagIndex := slices.IndexFunc(volume.UserTags, func(tag string) bool {
+			return cloud.ValidateTaskRunId(tag) == nil
+		})
+		if volumeTagIndex == -1 {
+			err := errors.New("failed to find a valid TaskRun ID")
+			log.Error(err, "appending to no TaskRun list anyway...", "instanceID", *instance.ID)
+			break
+		}
```

Comment:
this seems a typo to me. 
```suggestion
		if volumeTagIndex == -1 {
			err := errors.New("failed to find a valid TaskRun ID")
			log.Error(err, "appending to no TaskRun list anyway...", "instanceID", *instance.ID)
			instancesWithoutTaskRuns = append(instancesWithoutTaskRuns, *instance.ID)
			continue
		}
```